File: Classes/ProductClass.js
class Product{

    constructor(owner, availability, price= 1000){
        this.owner= owner;
        this.price= price;
        this.availability= availability;
        this.condition= "";
        this.description= "";
    }
    

    //Getters
    getOwner(){return this.owner;}
    getPrice(){return this.price;}
    getAvailability(){return this.availability;}
    getDescription(){return this.description;}
    getCondition(){return this.condition;}

    //Setters
    setOwner(owner){this.owner= owner;}
    setPrice(price){this.price= price;}
    setAvailability(availability){this.availability= availability;}
    setDescription(description){this.description= description;}
    setCondition(condition){this.condition= condition;}
    
}
module.exports.Product= Product;

File: lib/Classes/ProductClass.dart
class Product{
  String owner;
  double price;
  String availability;
  String condition;
  String description;
  String title;


  //Constructor
  Product({
    required this.owner,
    required this.availability,
    required this.price,
    this.condition = "UNKNOWN",
    this.description = "",
    required this.title,
  });

  //Getters
  String getOwner() => owner;
  double getPrice() => price;
  String getAvailability() => availability;
  String getCondition() => condition;
  String getDescription() => description;
  String getTitle() => title;

  //Setters
  void setOwner(String owner) => this.owner = owner;
  void setPrice(double price) => this.price = price;
  void setAvailability(String availability) => this.availability = availability;
  void setCondition(String condition) => this.condition = condition;
  void setDescription(String description) => this.description = description;
  void setTitle(String title) => this.title = title;

}